match/match.py

```diff
@@ -0,0 +1,81 @@
+import json
+
+# Get the userID and University name
+USER_ID = "UID3"  # TODO: connect to db
+USER_UNI = "UCLA"  # TODO: connect to db
+
+
+def main():
+    # open the file
+    with open("data.json", "r") as read_file:  # TODO: connect to db
+        data = json.load(read_file)
+
+    # provided data from the user: interests, name, [previous matches]
+    user_data = data["Universities"][USER_UNI][USER_ID]
+
+    # all users of the university
+    uni_users = data["Universities"][USER_UNI]
+
+    # Keep track of points for each user
+    match_points = {}
+
+    print(user_data["PreviousMatches"])
+
+    max_user = -1
+    user_ids = []
+    for i in uni_users:
+        max_user += 1
+        user_ids.append(i)
+    print(max_user)
+
+    """
+    This function loops over the data of user's university and 
+    assigns points for each of the other users 
+    """
+
+    def match():
```

Comment:
I'm not a huge fan of functions within functions because it makes it a little hard to read, but if they work for you then by all means